### ISSUE TITLE
Integrate AMP validation

### ISSUE THREAD

**Issue by pbakaus** [sonarwhal/sonar#545]
Would be great and probably fairly straightforward to integrate AMP validation into sonar, using the [amphtml-validator](https://www.npmjs.com/package/amphtml-validator) on npm.

**Comment by molant** [comment]
Hi @pbakaus 
trying to implement this but I'm having a bit of a problem with some of the tests. From what I read in the documentation there are [deprecation errors](https://www.ampproject.org/docs/reference/validation_errors#deprecation-errors). Those messages seem to be just warnings:

> This warning occurs when a previously valid AMP tag is found in the AMP document. It's only a warning; AMP documents with warnings continue to be valid. Currently no deprecated tags exist; the warning is reserved for future deprecations.

> This warning occurs when a previously valid AMP attribute is found in the AMP document. It's only a warning; AMP documents with warnings continue to be valid.
Identify deprecated attributes for each AMP tag by searching for deprecation in the AMP validator specification.

I'm giving the option to the user to decide if they want to fail in those warnings or not but I can't seem to get the validator to return anything. I'm currently using the `amp-video` element with the `muted autoplay` attributes. AFAIK, [the `mute` attribute is deprecated](https://www.ampproject.org/docs/reference/components/amp-video#muted-(deprecated)).

Here goes the html I'm using:
```html
<!doctype html>
<html amp lang="en">
  <head>
    <meta charset="utf-8">
    <script async src="https://cdn.ampproject.org/v0.js"></script>
    <script async custom-element="amp-video" src="https://cdn.ampproject.org/v0/amp-video-0.1.js"></script>
    <title>Hello, AMPs</title>
    <link rel="canonical" href="http://example.ampproject.org/article-metadata.html">
    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
    <script type="application/ld+json">
      {
        "@context": "http://schema.org",
        "@type": "NewsArticle",
        "headline": "Open-source framework for publishing content",
        "datePublished": "2015-10-07T12:02:41Z",
        "image": [
          "logo.jpg"
        ]
      }
    </script>
    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  </head>
  <body>
    <amp-video width="480"
    height="270"
    src="/video/tokyo.mp4"
    poster="/img/tokyo.jpg"
    layout="responsive"
    controls
    muted
    autoplay>
    <div fallback>
      <p>Your browser doesn't support HTML5 video.</p>
    </div>
    <source type="video/webm"
      src="/video/tokyo.webm">
  </amp-video>
  </body>
</html>
```

What am I doing wrong? Should I be testing for something else?

Thanks!

**Comment by powdercloud** [comment]
Hi @molant, I think what's going on here is that the mute attribute is documented as deprecated, but in our validator specification file it's not tagged as deprecated. This is a bug.
An easy way to get examples of our error / warnings output is to look at the tests. E.g. this one:
https://github.com/ampproject/amphtml/blob/master/validator/testdata/feature_tests/deprecation_warnings_and_errors.html
https://github.com/ampproject/amphtml/blob/master/validator/testdata/feature_tests/deprecation_warnings_and_errors.out
The first two lines in the output are deprecation warnings. The third is an error. So if you take this html and remove the disallowed tag from its body you'll have a working example for a deprecation warning. You can test this at validator.ampproject.org by copy/pasting into there and editing and so on. Hope this helps.

**Comment by molant** [comment]
Thanks @powdercloud !

PR has been updated. Any change @pbakaus or you could look into the [documentation part](https://github.com/sonarwhal/sonar/pull/547/files#diff-f3de84ab28143271b40546a5d466bea6) to make sure it's accurate?

**Comment by pbakaus** [comment]
All good from my end!
